corehq/form_processor/backends/couch/processor.py

```diff
@@ -173,7 +175,7 @@ def get_cases_from_forms(case_db, xforms):
         # don't process error forms which are being deprecated since they were never processed in the first place.
         # see http://manage.dimagi.com/default.asp?243382
         filtered_sorted_forms = [form for form in sorted_forms if not (form.is_deprecated and form.problem)]
-        touched_cases = {}
+        touched_cases = OrderedDict()
```

Comment:
I don't think we want to keep this because `OrderedDict` maintains an extra list for ordering so memory efficiency is worse. There may be some other operations that perform worse as well, but I have not evaluated it completely. I think this was only for the purpose of demonstrating the bug in Python 2, right @nickpell?